### PR TITLE
Update my.js and index.js

### PR DESCRIPTION
**Pull request by vaishnav98**
It is not required to call the bone.getPinObject(pin) twice , sufficient to return the stored object 'pinObject'

Edit in index.js , will that work for analogWrite() on DOUT ?

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -429,6 +429,21 @@ f.analogWrite = function(pin, value, freq, callback) {
 
     // Make sure the pin has a PWM associated
     if(typeof pin.pwm == 'undefined') {
+          if(typeof pin.gpio != 'undefined' ){
+            if(value>=0.5)
+            {
```
Comment (jadonk) [inline src/index.js]: Let's try to align on style. We probably need to run everything through js beautifier. As of now, please keep curly braces on the same line as the opening statement. Also, please include a space before the opening brace.

File: src/index.js [hunk 2]
```diff
@@ -429,6 +429,21 @@ f.analogWrite = function(pin, value, freq, callback) {
...
+                if(callback) callback(resp);
+                return(true);
+            }
+            else
+            {
+                resp=f.digitalWrite(pin,g.LOW,callback);
+                if(callback) callback(resp);
+                return(true);
+            }
+
```
Comment (jadonk) [inline src/index.js]: Please don't add blank lines without a purpose.

File: src/index.js (hunk 1 above)
Comment (vaishnav98) [reply src/index.js]: I have updated the patch by correcting the formatting issues , i did not notice these standards earlier , i will definitely make sure that i follow these formats in future.

File: src/index.js (hunk 2 above)
Comment (vaishnav98) [reply src/index.js]: Sorry, this was a mistake won't happen again

**Comment by vaishnav98** [comment]
It was a mistake from my side that i did not include a comment above the patch i submitted, I did not test the edit on actual hardware yet, i am going through the test subdirectory , are you looking for something like this : https://learn.adafruit.com/the-well-automated-arduino-library/travis-ci . I am not sure whether something like this is possible : https://www.pololu.com/blog/654/continuous-testing-for-arduino-libraries-using-platformio-and-travis-ci

File: src/index.js (hunk 2 above)
Comment (jadonk) [reply src/index.js]: No worries. Just using this as an early opportunity to discuss style.

RIght now, it looks like there is a tabs issue. I believe the files are using 4 spaces for tabs. Can you double-check your tabs vs. spaces in the pull request?

**Comment by jadonk** [comment]
I don't understand the comment regarding the specific link. We already have travis-ci integration. Unfortunately, there's no on-target execution.

We might want to add a js-beautify test case. We'll need to run js-beautify on everything first. Shall I take that on?

**Comment by vaishnav98** [comment]
I did not fully understand what you meant by "create something that  will make some of the code covered within travis-ci" that was why i searched the internet and found something similar that i posted here.I need to learn more about Travis-CI ,please excuse my ignorance.

It would be nice to run js-beautify on the entire code and add the test case.

File: src/index.js [hunk 3]
```diff
@@ -429,6 +429,18 @@ f.analogWrite = function(pin, value, freq, callback) {
 
     // Make sure the pin has a PWM associated
     if(typeof pin.pwm == 'undefined') {
+        if(typeof pin.gpio != 'undefined' ) {
+            if(value>=0.5) {
+                resp=f.digitalWrite(pin,g.HIGH,callback);
+                if(callback) callback(resp);
+                return(true);
+            }
+            else {
```
Comment (jadonk) [inline src/index.js]: I think the 2 lines above probably go together to match the style. I'm going to go ahead and merge for now and work on beautifying the whole set of source code before enforcing any particular style.

Comment (vaishnav98) [reply src/index.js]: Thank you :)

**Comment by vaishnav98** [comment]
I tested the patch,it works without any issues :)

**Comment by vaishnav98** [comment]
https://github.com/beagleboard/bonescript/issues/17

**Pull request merged**